workers/utils/searchUtils.js

```diff
@@ -79,115 +80,67 @@ export const getPubSearchData = (pubIds) => {
 				required: true,
 			},
 		],
-	})
-		.then((pubs) => {
-			const branchesToSync = [];
-			pubs.forEach((pubData) => {
-				const pub = pubData.toJSON();
-				const authorByline = pub.attributions
-					.map((attribution) => {
-						if (attribution.user) {
-							return attribution;
-						}
-						return {
-							...attribution,
-							user: {
-								id: attribution.id,
-								fullName: attribution.name,
-							},
-						};
-					})
-					.filter((attribution) => {
-						return attribution.isAuthor;
-					})
-					.sort((foo, bar) => {
-						if (foo.order < bar.order) {
-							return -1;
-						}
-						if (foo.order > bar.order) {
-							return 1;
-						}
-						if (foo.createdAt < bar.createdAt) {
-							return 1;
-						}
-						if (foo.createdAt > bar.createdAt) {
-							return -1;
-						}
-						return 0;
-					})
-					.map((author, index, array) => {
-						const separator =
-							index === array.length - 1 || array.length === 2 ? '' : ', ';
-						const prefix = index === array.length - 1 && index !== 0 ? ' and ' : '';
-						return `${prefix}${author.user.fullName}${separator}`;
-					})
-					.join('');
+	});
 
-				const managerIds = pub.managers.map((manager) => {
-					return manager.userId;
-				});
+	const branchesToSync = [];
+	for (let index = 0; index < pubs.length; index++) {
+		const pub = pubs[index].toJSON();
+		const authorByline = generatePlainAuthorString(pub);
+		// eslint-disable-next-line no-await-in-loop
+		const scopeData = await getScope({ pubId: pub.id, communityId: pub.community.id });
+		// eslint-disable-next-line no-await-in-loop
+		const { members } = await getMembers({ scopeData: scopeData });
+		const accessIds = members.map((member) => {
+			return member.userId;
+		});
+		pub.branches.forEach((branch) => {
+			/* Assume metadata is 3000 characters = 3000 bytes */
+			const data = {
+				pubId: pub.id,
+				title: pub.title,
+				slug: pub.slug,
+				avatar: pub.avatar,
+				description: pub.description,
+				byline: authorByline ? `by ${authorByline}` : '',
+				communityId: pub.community.id,
+				communityDomain: pub.community.domain || `${pub.community.subdomain}.pubpub.org`,
+				communityTitle: pub.community.title,
+				communityAccentColorLight: pub.community.accentColorLight,
```

Comment:
I think we can drop the properties from here that aren't slugs or natural language strings.

Reply:
Our search cycle doesn't hit our postgres DB at all. The client directly queries algolia for search results.  Anything that we want to render in the results has to be stored in algolia records.